client/src/components/ElementEditor/Element.js

```diff
@@ -53,6 +74,68 @@ class Element extends Component {
         captureDraggingState: true,
       });
     }
+    // Check if formSchema state has already been loaded before opening a block
+    // This can happen if there was a validation error on a block after performing a Page save
+    if (this.props.formStateExists) {
+      this.setState({
+        formHasRendered: true
+      });
+    }
+  }
+
+  componentDidUpdate() {
```

Comment:
It's the only way I could work out how to do this.

Essentially we need to ensure that several dependent occurred in a sequence sending a publish request, e.g. when publishing
- The form has rendered before sending a request so that validation can occur - it won't have rendered if the element has been toggled open
- If the form is dirty, save the form first to trigger validation. If it's valid then publish.

Reply:
Will be addressed in follow up card https://github.com/silverstripe/silverstripe-elemental/issues/1164